packages/web/src/components/cells/hover.tsx

```diff
@@ -0,0 +1,143 @@
+import {
+  CodeCellType,
+  TsServerQuickInfoRequestPayloadType,
+  TsServerQuickInfoResponsePayloadType,
+  TsServerQuickInfoResponseType,
+} from '@srcbook/shared';
+import { Extension, hoverTooltip } from '@uiw/react-codemirror';
+import { mapCMLocationToTsServer } from './util';
+import { SessionChannel } from '@/clients/websocket';
+import { createRoot } from 'react-dom/client';
+import { useEffect, useState } from 'react';
+
+/** Hover extension for TS server information */
+export function tsHover(sessionId: string, cell: CodeCellType, channel: SessionChannel): Extension {
+  return hoverTooltip(async (view, pos) => {
+    if (cell.language !== 'typescript') {
+      return null; // bail early if not typescript
+    }
+
+    const { from, to, text } = view.state.doc.lineAt(pos);
+    let start = pos;
+    let end = pos;
+
+    while (start > from && /\w/.test(text[start - from - 1])) start--;
```

Comment:
Is this regex sufficient for all symbols that we might hover over?
(I think so... just checking)

Reply:
yep it's just for calculating where to snag the quickInfo and where to render the tooltip from, nothing else